### PR TITLE
Fixes #84: Pressing Enter on search box refreshed the page

### LINKED ISSUE #84: Ignore Enter in the search box
Currently when one press Enter in the search box, the site reloads. 
IMHO enter should just be ignored.

### PR DESCRIPTION
**Pull request by AmadeusW**
Removing the form tag solves the issue.
This PR also removes the form from basic layout (for home page, upload page) where there is no code to search through.

### CODE REVIEW

File: src/SourceBrowser.Site/Views/Shared/_BrowseLayout.cshtml [hunk 1]
```diff
@@ -19,12 +19,10 @@
...
             <div class="input-element place-right" id="search-form">
-                <form>
-                    <div class="input-control text">
-                        <input type="text" id="search-box" placeholder="Search...">
-                        <button type="submit" id="search-button"></button>
-                    </div>
-                </form>
+                <div class="input-control text">
+                    <input type="text" id="search-box" placeholder="Search...">
+                    <button type="submit" id="search-button"></button>
```
Comment (JoshVarty) [inline src/SourceBrowser.Site/Views/Shared/_BrowseLayout.cshtml]: This probably shouldn't be of type submit if it's not in the form anymore. I think you can probably just get rid of the type.

File: src/SourceBrowser.Site/Scripts/search.js [hunk 2]
```diff
@@ -2,11 +2,7 @@
     var path = window.location.pathname;
     var splitPath = path.split('/');
     splitPath = splitPath.filter(function (v) { return v !== '' });
-    if (splitPath.length <= 2)
-    {
-        $("#search-form").hide();
```
Comment (JoshVarty) [inline src/SourceBrowser.Site/Scripts/search.js]: Yeah this was bad. But how does the search box get shown now? Is it visible on all pages?

Comment (AmadeusW) [inline src/SourceBrowser.Site/Scripts/search.js]: It's physically gone from `_Layout.cshtml`. The form is only present in *_BrowseLayout.cshtml`

File: src/SourceBrowser.Site/Views/Shared/_BrowseLayout.cshtml (hunk 1 above)
Comment (AmadeusW) [inline src/SourceBrowser.Site/Views/Shared/_BrowseLayout.cshtml]: Good call. I've removed the type. While testing if the button still works, found out that it never worked. But that's fixed now, too :)

File: src/SourceBrowser.Site/Scripts/search.js (hunk 2 above)
Comment (JoshVarty) [inline src/SourceBrowser.Site/Scripts/search.js]: Sounds good.

**Pull request merged**